llvm/lib/CodeGen/MachineCopyPropagation.cpp

```diff
@@ -885,6 +885,10 @@ void MachineCopyPropagation::ForwardCopyPropagateBlock(MachineBasicBlock &MBB) {
       assert(!Reg.isVirtual() &&
              "MachineCopyPropagation should be run after register allocation!");
 
+      // Skip invalidating constant registers.
+      if (MRI->isReserved(Reg) && MRI->isConstantPhysReg(Reg))
```

Comment:
Don't need the reserved check?

Reply:
This is only relevant for defs, so move this down inside the def handling?

Reply:
Removed reserved check and moved `isConstantPhysReg` check inside the def handling.